### PR TITLE
[liblas] Add new feature gdal

### LINKED ISSUE #16005: [libLAS] GDAL support
****

The pointclouds I work with have specific requirements for georeferencing; when using libLAS I need its optional WITH_GDAL feature enabled (it's off by default); however the vcpkg port doesn't provide this option. It would certainly be nice though.

**Proposed solution**
Please add an option for GDAL support (libLAS already provides a WITH_GDAL option, I *think* this could be exposed as a vcpkg port option)

### PR DESCRIPTION
**Pull request by longhuan2018**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #16005
  Merge PR #15522

Fix find_package(liblas),  libLAS_INCLUDE_DIRS path incorrect bug.
liblas-config.cmake is stored in the share\liblas directory,  So the relative path should be ../.. instead of ..
```
get_filename_component (PROJECT_ROOT_DIR "${_DIR}/.." ABSOLUTE)
set (libLAS_INCLUDE_DIRS "${PROJECT_ROOT_DIR}/include")
set (libLAS_LIBRARY_DIRS "${PROJECT_ROOT_DIR}/lib")
set (libLAS_BINARY_DIRS "${PROJECT_ROOT_DIR}/bin")
```

libLAS_LIBRARIES contains  geotiff_library,  we need to add  find_package (geotiff CONFIG REQUIRED)  to the liblas-config.cmake  file

### CODE REVIEW

File: ports/liblas/CONTROL [hunk 1]
```diff
@@ -1,13 +1,10 @@
 Source: liblas
 Version: 1.8.1
-Port-Version: 6
-Build-Depends: boost, boost-thread, boost-system, boost-iostreams, boost-filesystem, boost-detail, libgeotiff
+Port-Version: 7
+Build-Depends: boost, boost-thread, boost-system, boost-iostreams, boost-filesystem, boost-detail, libgeotiff, libjpeg-turbo, zlib
```
Comment (dg0yt) [inline ports/liblas/CONTROL:4]: The PR is named "support gdal". The new core dependencies are not required for this purpose: You might use feature dependencies (at least after converting `CONTROL` to `vcpkg.json` with `vcpkg format-manifest  ports/liblas/CONTROL`).

File: ports/liblas/portfile.cmake [hunk 2]
```diff
@@ -13,20 +13,25 @@ vcpkg_extract_source_archive_ex(
...
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
-    OPTIONS 
+    OPTIONS ${FEATURE_OPTIONS}
         -DBUILD_OSGEO4W=OFF # Disable osgeo4w
         -DWITH_TESTS=OFF
         -DWITH_UTILITIES=OFF
-        -DCMAKE_DISABLE_FIND_PACKAGE_ZLIB=${CMAKE_DISABLE_FIND_PACKAGE_ZLIB}
-        -DCMAKE_DISABLE_FIND_PACKAGE_JPEG=${CMAKE_DISABLE_FIND_PACKAGE_JPEG}
```
Comment (dg0yt) [inline ports/liblas/portfile.cmake:29]: If keeping jpeg and zlib features, these might be handled via `vcpkg_check_features` as inverted feature options.

File: ports/liblas/CONTROL (hunk 1 above)
Comment (dg0yt) [reply ports/liblas/CONTROL:4]: OTOH, libgeotiff pulls in libjpeg-turbo and zlib dependencies anyway. So there is nothing gained from optional features, and CMAKE_DISABLE_FIND_PACKAGE_... is not desirable. 
It just isn't clear from the PR structure and comments.

File: ports/liblas/portfile.cmake (hunk 2 above)
Comment (longhuan2018) [reply ports/liblas/portfile.cmake:29]: JPEG and zlib are the dependencies of libgeotif, so it is not necessary to be a feature of liblas  (and the CMAKE_DISABLE_FIND_PACKAGE_ZLIB,  CMAKE_DISABLE_FIND_PACKAGE_JPEG option is not used in liblas)

File: ports/liblas/CONTROL [hunk 3]
```diff
@@ -1,13 +1,10 @@
 Source: liblas
 Version: 1.8.1
-Port-Version: 6
+Port-Version: 7
 Build-Depends: boost, boost-thread, boost-system, boost-iostreams, boost-filesystem, boost-detail, libgeotiff
 Description: A C/C++ library for reading and writing the very common LAS LiDAR format.
+Default-Features: gdal
```
Comment (dg0yt) [inline ports/liblas/CONTROL:6]: IMO this shouldn't be a default feature, given that the combination of GDAL 3.x and liblas is not supported upstream. (In fact, upstream recommends switching to pdal.) 
I understant that this is needed to get CI coverage for this PR now. But it also adds a maintenance burden to future changes, without upstream support.

Comment (longhuan2018) [reply ports/liblas/CONTROL:6]: When the CI compilation passes, I will delete the default value

**Comment by JackBoosY** [comment]
Can you please resolve the file conflicts?

**Comment by JackBoosY** [comment]
This PR has been inactive for a long time, please reopen it if there is any progress.

**Pull request closed without merging**